docs/how-to-guides/upgrade-bootstrap-kubelet.md

```diff
@@ -0,0 +1,85 @@
+# Upgrading bootstrap Kubelet
+
+## Contents
+
+- [Introduction](#introduction)
+- [Steps](#steps)
+  - [Step 1: Drain the node](#step-1-drain-the-node)
+  - [Step 2: Find out the IP and SSH](#step-2-find-out-the-ip-and-ssh)
+  - [Step 3: Upgrade Kubelet on node](#step-3-upgrade-kubelet-on-node)
+  - [Step 4: Verify](#step-4-verify)
+- [Caveats](#caveats)
+
+## Introduction
+
+[Kubelet](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) is a daemon
+that runs on every node and it is responsible for managing Pods on the node.
+
+Lokomotive cluster runs two different sets of Kubelet processes. Initially, **bootstrap** Kubelet
+configured on the node as a `rkt` pod joins the cluster, and then Kubelet pod managed using
+DaemonSet(self-hosted Kubelet) takes over the bootstrap Kubelet. Self-hosted Kubelet allows seamless
+updates between Kubernetes patch version for nodes and also to tune nodes configuration using tools
+like `kubectl`.
+
+At the time of writing `lokoctl` cannot update bootstrap Kubelet, so this document explains how to
+perform this operation manually.
+
+## Steps
+
+Perform the following steps on each node, one node at a time.
+
+### Step 1: Drain the node
+
+> **Caution:** If you are using a local directory as a storage for a workload, it will be disturbed
+> by this operation. To avoid this move the workload to another node and let the application
+> replicate the data. If the application does not support data replication across instances, then
+> expect downtime.
+
+```bash
+kubectl drain --ignore-daemonsets <node name>
+```
+
+### Step 2: Find out the IP and SSH
+
+Find the IP of the node by visiting the cloud provider dashboard.
```

Comment:
I'd expect something like: `Then, connect to selected machine using SSH`.

Maybe instead of SSH, we could do this using privileged DaemonSet? Or using something like https://github.com/kvaps/kubectl-node-shell ?

Reply:
Added that line. About using kubectl node-shell, I am not sure that should be a team decision to add this tool. Does it work without PSPs? What service account does it use? With webhook server if it uses default service account then it might fail, no?

Reply:
>Added that line. About using kubectl node-shell, I am not sure that should be a team decision to add this tool. Does it work without PSPs? What service account does it use? With webhook server if it uses default service account then it might fail, no?

I'd expect it to spawn the pod directly on the selected node.

Reply:
Oh, I wonder if the shell connection breaks if you restart the kubelet :smile: It might not be a good idea then.